python/ray/serialization.py

```diff
@@ -42,6 +44,62 @@
 
 logger = logging.getLogger(__name__)
 
+_pickle_whitelist = None
+
+
+class RestrictedUnpickler(pickle.Unpickler):
+    def find_class(self, module, name):
+        if _pickle_whitelist is None or (
+            module in _pickle_whitelist
+            and (_pickle_whitelist[module] is None or name in _pickle_whitelist[module])
+        ):
+            return super().find_class(module, name)
+
+        if module == "ray.serialization":
+            return super().find_class(module, name)
+
+        # Forbid everything else.
+        raise pickle.UnpicklingError("global '%s.%s' is forbidden" % (module, name))
+
+
+def restricted_loads(
+    serialized_data,
+    *,
+    fix_imports=True,
+    encoding="ASCII",
+    errors="strict",
+    buffers=None,
+):
+    if isinstance(serialized_data, str):
+        raise TypeError("Can't load pickle from unicode string")
+    file = io.BytesIO(serialized_data)
+    return RestrictedUnpickler(
+        file, fix_imports=fix_imports, buffers=buffers, encoding=encoding, errors=errors
+    ).load()
+
+
+def patch_pickle_for_security():
+    global _pickle_whitelist
+    whitelist_path = ray_constants.RAY_PICKLE_WHITELIST_CONFIG_PATH
+    if whitelist_path is None:
+        return
+
+    _pickle_whitelist = yaml.safe_load(open(whitelist_path, "rt")).get(
+        "pickle_whitelist", None
+    )
+    if _pickle_whitelist is None:
+        return
+
+    if "*" in _pickle_whitelist:
+        _pickle_whitelist = None
+    for module, attr_list in _pickle_whitelist.items():
+        if "*" in attr_list:
+            _pickle_whitelist[module] = None
+    pickle.loads = restricted_loads
+
+
+patch_pickle_for_security()
```

Comment:
Can we move this logic to a new file like `python/ray/_private/security_whitelist.py` or `python/ray/_private/security.py`

Reply:
done